### ISSUE TITLE
Add syntax highlighting to method input field in Schema Editor.

### ISSUE THREAD

**Issue by cmorgner** [structr/structr#308]
It would be really helpful to have some kind of syntax highlighting in the Schema Editor, making the creation of complex function expressions easier.

**Comment by vigorouscoding** [comment]
IIRC this issue was supposed to be resolved by replacing the textareas in the schema editor with code mirror.
This has already been done for FunctionProperty but remains to be done for custom schema methods.

Syntax highlighting might be a problem though since the content can be JS or StructrScript.
I will look into finding a way to auto-detect the flavor of the code.
Also, StructrScript will need its custom syntax highlighter which might be a problem. I will look into that as well.

**Comment by vigorouscoding** [comment]
I added CodeMirror functionality to the schema methods dialog (commit 8a26aa8cddac123c08732a6e82b67c290cf94ad3).

I added auto-sensing for JavaScript and we could try and create a language sensing for StructrScript by extending the CodeMirror functionality (see https://codemirror.net/doc/manual.html#modeapi)

For now I would guess that simply adding CodeMirror should suffice.

The only problem is that CodeMirror automatically adds 30px padding at the bottom so scrolling can become an annoyance - but a little more investigation in the CSS should fix that.
